contracts/TokenVesting.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "./Governable.sol";
+import "./interface/ISOLACE.sol";
+import "./interface/ITokenVesting.sol";
+
+/**
+ * @title TokenVesting
+ * @author solace.fi
+ * @notice Stores and handles vested [**SOLACE**](./SOLACE) tokens for SOLACE investors.
+ *
+ * Predetermined agreement with investors for a linear unlock over three years, with a six month cliff.
+ * We use a Unix timestamp of 1638209176 for the vestingStart, using the following transaction as our reference - https://etherscan.io/tx/0x71f1de15ee75f414c454aec3612433d0123e44ec5987515fc3566795cd840bc3
+ */
+
+ contract TokenVesting is ITokenVesting, ReentrancyGuard, Governable {
+ 
+    /***************************************
+    GLOBAL VARIABLES
+    ***************************************/
+
+    /// @notice SOLACE Token.
+    address public override solace;
+
+    /// @notice timestamp that investor tokens start vesting.
+    uint256 public override vestingStart;
+
+    /// @notice timestamp that cliff for investor tokens finishes.
+    uint256 public override cliff;
+
+    /// @notice timestamp that investor tokens finish vesting.
+    uint256 public override vestingEnd;
```

Comment:
You can make "vestingStart" and "vestingEnd" immutable to save gas.

Reply:
Nice did this